internal/engine/experiment/quicping/quicping.go

```diff
@@ -0,0 +1,479 @@
+	IsAnomaly bool `json:"-"`
+}
+
+// GetSummaryKeys implements model.ExperimentMeasurer.GetSummaryKeys.
+func (m *Measurer) GetSummaryKeys(measurement *model.Measurement) (interface{}, error) {
+	return SummaryKeys{IsAnomaly: false}, nil
+}
+
+// buildHeader creates the unprotected QUIC header.
+// https://www.rfc-editor.org/rfc/rfc9000.html#name-initial-packet
+func buildHeader(destConnID, srcConnID ConnectionID, payloadLen int) []byte {
+	hdr := []byte{0xc3} // long header type, fixed
+
+	version := make([]byte, 4)
+	binary.BigEndian.PutUint32(version, uint32(0xbabababa))
+	hdr = append(hdr, version...) // version
+
+	lendID := uint8(len(destConnID))
+	hdr = append(hdr, lendID)        // destination connection ID length
+	hdr = append(hdr, destConnID...) // destination connection ID
+
+	lensID := uint8(len(srcConnID))
+	hdr = append(hdr, lensID)       // source connection ID length
+	hdr = append(hdr, srcConnID...) // source connection ID
+
+	hdr = append(hdr, 0x0) // token length
+
+	remainder := 4 + payloadLen
+	remainder_mask := 0b100000000000000
+	remainder_mask |= remainder
+	remainder_b := make([]byte, 2)
+	binary.BigEndian.PutUint16(remainder_b, uint16(remainder_mask))
+	hdr = append(hdr, remainder_b...) // remainder length: packet number + encrypted payload
+
+	pn := make([]byte, 4)
+	binary.BigEndian.PutUint32(pn, uint32(2))
+	hdr = append(hdr, pn...) // packet number
+
+	return hdr
+}
+
+// buildPacket constructs an Initial QUIC packet
+// and applies Initial protection.
+// https://www.rfc-editor.org/rfc/rfc9001.html#name-client-initial
+func buildPacket() ([]byte, ConnectionID, ConnectionID) {
+	destConnID, srcConnID := generateConnectionIDs()
+	// generate random payload
+	minPayloadSize := 1200 - 14 - (len(destConnID) + len(srcConnID))
+	randomPayload := make([]byte, minPayloadSize)
+	rand.Read(randomPayload)
+
+	clientSecret, _ := computeSecrets(destConnID)
+	encrypted := encryptPayload(randomPayload, destConnID, clientSecret)
+	hdr := buildHeader(destConnID, srcConnID, len(encrypted))
+	raw := append(hdr, encrypted...)
+
+	raw = encryptHeader(raw, hdr, clientSecret)
+	return raw, destConnID, srcConnID
+}
+
+// generateConnectionID generates a connection ID using cryptographic random
+func generateConnectionID(len int) ConnectionID {
+	b := make([]byte, len)
+	_, err := rand.Read(b)
+	runtimex.PanicOnError(err, "rand.Read failed")
+	return ConnectionID(b)
+}
+
+// generateConnectionIDForInitial generates a connection ID for the Initial packet.
+// It uses a length randomly chosen between 8 and 18 bytes.
+func generateConnectionIDForInitial() ConnectionID {
+	r := make([]byte, 1)
+	_, err := rand.Read(r)
+	runtimex.PanicOnError(err, "rand.Read failed")
+	len := minConnectionIDLenInitial + int(r[0])%(maxConnectionIDLen-minConnectionIDLenInitial+1)
+	return generateConnectionID(len)
+}
+
+// generateConnectionIDs generates a destination and source connection ID.
+func generateConnectionIDs() ([]byte, []byte) {
```

Comment:
Please, move this function to a separate file. Logically, this is a basic building block for the main implementation of the experiment, so I'd like it to be in a separate file. The way I see it, our stack here is like:

```
  experiment (quicping.go)
  --------
  separate file (maybe quic.go?)
  --------
  crypto (crypto.go)

```